cmd/fieldnamedocscheck/field_name_docs_check.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"unicode"
+
+	flag "github.com/spf13/pflag"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+)
+
+var (
+	typeSrc = flag.StringP("type-src", "s", "", "From where we are going to read the types")
+)
+
+func main() {
+	flag.Parse()
+
+	if *typeSrc == "" {
+		klog.Fatalf("Please define -s flag as it is the api type file")
+	}
+
+	docsForTypes := kruntime.GetFieldAndDocFrom(*typeSrc)
+	rc := false
+
+	for _, ks := range docsForTypes {
+		structName := ks[""].Name
+		for field, p := range ks {
+			rc = checkFieldNameAndDoc(structName, field, p.Doc, ks)
+		}
+	}
+
+	if rc {
+		os.Exit(1)
+	}
+}
+
+func checkFieldNameAndDoc(structName, fieldName, doc string, fieldToPair map[string]kruntime.Pair) bool {
+	rc := false
+	visited := sets.Set[string]{}
+
+	words := strings.FieldsFunc(doc, func(r rune) bool {
+		return !unicode.IsDigit(r) && !unicode.IsLetter(r)
+	})
+
+	// The rule is to check if the doc contains any field name in this struct which is
+	// different from its tag name, and only check for the first occurrence for each field.
+	// To avoid false-positives, please don't use any field name in the doc.(But it's ok
```

Comment:
I don't think this is a reasonable restriction? My guess is that it is fairly common to need to be able to reference the abstract concept that the field is about and that's often the same word as the field name. And it is not obvious how to make an escape sequence that won't show up in the docs.

I'd rather e.g. require field names to be in back-ticks ``` `like this` ``` and then you can check everything in the back-ticks.

Reply:
The reason I added the restriction is I didn't find a good way to identify the field name in docs. And I really like the backticks, it makes life easier and some docs already used them, but if people forget to add backticks, we won't be able to detect the wrong name in docs.
Meanwhile, we have lots of docs that didn't use the back-ticks, we have to raise a ticket to update them, but still don't know how to verify if people added back-ticks correctly, maybe require at least one pair of backticks or just don't check them, any suggestions?

Reply:
I suggest adding a linter checking back-tick quoted names first (should have zero false positives).

Then a manual pass adding back-ticks to the doc strings.

I don't think it's helpful to even think about a linter for that if we haven't done a manual pass, because even if you could make a good linter it would basically flag everything, which isn't helpful.

Reply:
Thanks, updated the linter.